src/Umbraco.SampleSite.Website/Views/MacroPartials/LatestBlogposts.cshtml

```diff
@@ -1,25 +1,41 @@
 ﻿@using ContentModels = Umbraco.Web.PublishedContentModels;
 @using Umbraco.Web;
 @inherits Umbraco.Web.Macros.PartialViewMacroPage
-@{ 
+@{
     var startNodeId = Model.MacroParameters["startNodeId"] != null ? Model.MacroParameters["startNodeId"] : Model.Content.Id;
     var numberOfPosts = 3;
     if (Model.MacroParameters["numberOfPosts"] != null)
     {
         int.TryParse((string)Model.MacroParameters["numberOfPosts"], out numberOfPosts);
     }
- }
+
+
+}
 @if (startNodeId != null)
 {
     @* Get the starting page *@
     var startNode = Umbraco.TypedContent(startNodeId);
-    var blogposts = startNode.Children.OrderByDescending(x => x.CreateDate).Take(numberOfPosts);
+    //Gets all blogposts to calculate pages
+    var blogposts = startNode.Children.OrderByDescending(x => x.CreateDate);
```

Comment:
Only thing here is that you are not resolving the IEnumerable to a list or array which means it will re-iterate the enumerable for all of these methods like Count() and Any(), etc...  These should be resolved to a list and then the native `Count` methods used, Linq can cause inadvertent performance problems due to this particular circumstance. Using `Count()` when a native `Count` method is available will cause the full enumeration to occur again along with allocating new enumerators. Similarly for Any(), etc... Linq isn't smart and won't call into native methods even if they exist.

Reply:
Oh wow, thanks for this comment! I'll look into it for next time.